src/rotator_library/async_locks.py

```
@@ -0,0 +1,388 @@
+            if self._last_release is not None:
+                elapsed = time.time() - self._last_release
+                if elapsed >= self._min_interval * self._burst_size:
+                    self._burst_remaining = self._burst_size
+
+            self._burst_remaining -= 1
+
+    def release(self) -> None:
+        """Release the lock."""
+        self._last_release = time.time()
+
+    async def __aenter__(self):
```

Ruff (0.15.4)

[warning] Missing return type annotation for special method `__aenter__`

(ANN204)